src/python/views/demo_tigergraph_circle/__init__.py

```diff
@@ -0,0 +1,774 @@
+        .bind(edge_weight='amount')      \
+        .bind(source='from_id', destination='to_id') \
+        .bind(edge_title='amount', edge_label='amount') \
+        .nodes(nodes_df) \
+        .bind(node='n') \
+        .addStyle(bg={'color': 'white'}) \
+        .encode_point_color("color") \
+        .encode_edge_color("color") \
+        .encode_point_icon('type', categorical_mapping={'User': 'laptop',
+                                                        'Transaction': 'server'},
+                           default_mapping='question')
+
+
+# .encode_point_size('', ["blue", "yellow", "red"],  ,as_continuous=True)
+    # if not (node_label_col is None):
+    #     g = g.bind(point_title=node_label_col)
+
+    # if not (edge_label_col is None):
+    #     g = g.bind(edge_title=edge_label_col)
+
+    url = g.plot(render=False)
+
+    toc = time.perf_counter()
+    metrics['graphistry_time'] = toc - tic
+    logger.info(f'Graphisty Time: {metrics["graphistry_time"]}')
+    logger.info('Generated viz, got back urL: %s', url)
+
+    return url
+
+
+import pyTigerGraphBeta as tg
+import pandas as pd
+import datetime
+
+
+#
+# def flatten(lst_of_lst):
+#     try:
+#         if type(lst_of_lst[0]) == list:
+#             return [item for sublist in lst_of_lst for item in sublist]
+#         else:
+#             return lst_of_lst
+#     except:
+#         print('fail', lst_of_lst)
+#         return lst_of_lst
+#
+#
+
+
+# Given filter settings, generate/cache/return dataframes & viz
+@st.cache(suppress_st_warning=True, allow_output_mutation=True)
+def run_filters(user_id, conn):
+    global metrics
+    # global conn
+
+    # if conn is None:
+    #     conn = tg.TigerGraphConnection(host="https://fraud-streamlit.i.tgcloud.io", graphname="AntiFraud",
+    #                                    username="tigergraph", password="tigergraph")
+
+    # conn.getToken("tufp2os5skgljafj7ol4ikht2atc7rbj")
+    logger.info("Installing Queries")
+    res = conn.gsql(
+    '''
+    use graph {}
+    ls 
+    '''.format(conn.graphname), options=[])
+
+    ind = res.index('Queries:') + 1
+    installTran = True
+    for query in res[ind:]:
+        if 'totalTransaction' in query:
+            installTran = False
+    
+    if installTran:
+        conn.gsql(
+        '''
+        use graph AntiFraud
+        CREATE QUERY totalTransaction(Vertex<User> Source) FOR GRAPH AntiFraud {  
+            start = {Source};
+
+            transfer = SELECT tgt
+                FROM start:s -(User_Transfer_Transaction:e) - :tgt;
+
+            receive = select tgt
+                FROM start:s -(User_Recieve_Transaction:e) -:tgt;
+
+            PRINT transfer, receive;
+        }
+        Install query totalTransaction 
+        ''', options=[])
+
+    logger.info('Querying Tigergraph')
+    tic = time.perf_counter()
+
+    results = conn.runInstalledQuery("circleDetection", {"srcId": user_id}, sizeLimit=1000000000, timeout=120000)
```

Comment:
if it helps on the > 100 result graphs, i remember we saw speed issues with pyTigergraph, where pygraphistry's .gsql call was loading data something like 10-100X faster. this may be fixed in pytigergraph now tho? (i think this demo had some cool 10k-100k graphs)

Reply:
The new pyTigerGraph package used in the views uses a Python driver(Java free) which enhances the speed and performance of the queries.